src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -51,124 +45,120 @@ function forceUpdateIfMounted() {
  *
  * @see http://www.w3.org/TR/2012/WD-html5-20121025/the-input-element.html
  */
-var ReactDOMInput = ReactClass.createClass({
-  displayName: 'ReactDOMInput',
-  tagName: 'INPUT',
-
-  mixins: [AutoFocusMixin, LinkedValueUtils.Mixin, ReactBrowserComponentMixin],
-
-  getInitialState: function() {
-    var defaultValue = this.props.defaultValue;
-    return {
-      initialChecked: this.props.defaultChecked || false,
-      initialValue: defaultValue != null ? defaultValue : null
-    };
+var ReactDOMInput = {
+  getNativeProps: function(props, context) {
+    // TODO: AutoFocusMixin
+    // TODO: Check LinkedValueUtils.propTypes
+
+    var value = LinkedValueUtils.getValue(props);
+    var checked = LinkedValueUtils.getChecked(props);
+
+    var nativeProps = assign({}, props, {
+      defaultChecked: undefined,
+      defaultValue: undefined,
+      value: value != null ? value : this._wrapperState.initialValue,
+      checked: checked != null ? checked : this._wrapperState.initialChecked,
+      onChange: this._wrapperState.onChange
+    });
```

Comment:
This copying should be avoidable now that we don't have to adhere to a common pretty API but fine as a first step.